src/com/facebook/buck/jvm/groovy/GroovyTestDescription.java

```diff
@@ -0,0 +1,193 @@
+                                resolver.getAllRules(args.providedDeps.get()))),
+                        pathResolver.filterBuildRuleInputs(
+                            defaultJavacOptions.getInputs(pathResolver)))),
+                pathResolver,
+                args.srcs.get(),
+                ResourceValidator.validateResources(
+                    pathResolver,
+                    params.getProjectFilesystem(),
+                    args.resources.get()),
+                defaultJavacOptions.getGeneratedSourceFolderName(),
+                args.labels.get(),
+                args.contacts.get(),
+                Optional.<SourcePath>absent(),
+                new BuildTargetSourcePath(abiJarTarget),
+                /* additionalClasspathEntries */ ImmutableSet.<Path>of(),
+                args.testType.or(TestType.JUNIT),
+                stepFactory,
+                args.vmArgs.get(),
+                ImmutableMap.<String, String>of(),
+                validateAndGetSourcesUnderTest(
+                    args.sourceUnderTest.get(),
+                    params.getBuildTarget(),
+                    resolver),
+                Optional.<Path>absent(),
+                Optional.<String>absent(),
+                args.testRuleTimeoutMs.or(defaultTestRuleTimeoutMs),
+                args.getRunTestSeparately(),
+                args.stdOutLogLevel,
+                args.stdErrLogLevel,
+                testTempDirOverride));
+
+    resolver.addToIndex(
+        CalculateAbi.of(
+            abiJarTarget,
+            pathResolver,
+            params,
+            new BuildTargetSourcePath(test.getBuildTarget())));
+
+    return test;
+  }
+
+  public static ImmutableSet<BuildRule> validateAndGetSourcesUnderTest(
+      ImmutableSet<BuildTarget> sourceUnderTestTargets,
+      BuildTarget owner,
+      BuildRuleResolver resolver) {
+    ImmutableSet.Builder<BuildRule> sourceUnderTest = ImmutableSet.builder();
+    for (BuildTarget target : sourceUnderTestTargets) {
+      BuildRule rule = resolver.getRule(target);
+      if (!(rule instanceof JavaLibrary)) {
+        // In this case, the source under test specified in the build file was not a Java library
+        // rule. Since EMMA requires the sources to be in Java, we will throw this exception and
+        // not continue with the tests.
+        throw new HumanReadableException(
+            "Specified source under test for %s is not a Java library: %s (%s).",
+            owner,
+            rule.getFullyQualifiedName(),
+            rule.getType());
+      }
+      sourceUnderTest.add(rule);
+    }
+    return sourceUnderTest.build();
+  }
+
+  @SuppressFieldNotInitialized
+  public static class Arg extends GroovyLibraryDescription.Arg {
+    public Optional<ImmutableSortedSet<String>> contacts;
+    public Optional<ImmutableSortedSet<Label>> labels;
+    @Hint(isDep = false) public Optional<ImmutableSortedSet<BuildTarget>> sourceUnderTest;
```

Comment:
I'm translating this as: 

- remove the part of the `Arg` that refers to it.
- destroy `validateAndGetSourcesUnderTest`
- pass an empty set to `JavaTest`'s constructor arg.

Is there a plan to remove it from the java side as well?